Calo/Calo/Views/CaloTableViewController.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  CaloTableViewController.swift
+//  Calo
+//
+//  Created by Gladymir Philippe on 10/11/20.
+//  Copyright © 2020 Gladymir Philippe. All rights reserved.
+//
+
+import UIKit
+import CoreData
+import SwiftChart
+
+class CaloTableViewController: UITableViewController {
+    
+    @IBOutlet private weak var chartView: Chart!
+
+        // MARK: - Properties
+        let caloriesController = CaloriesController()
+
+        lazy var fetchedResultsController: NSFetchedResultsController<CalorieEntry> = {
+            let fetchRequest: NSFetchRequest<CalorieEntry> = CalorieEntry.fetchRequest()
+            fetchRequest.sortDescriptors = [NSSortDescriptor(key: "date",
+                                                             ascending: true),
+                                            NSSortDescriptor(key: "date",
+                                                             ascending: true)
+            ]
+            let moc = CoreDataStack.shared.mainContext
+            let frc = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: moc, sectionNameKeyPath: nil, cacheName: nil)
+
+            frc.delegate = self
+
+            do {
+                try frc.performFetch()
+            } catch {
+                print("Error fetching")
+            }
+            return frc
+        }()
+
+        // MARK: - Lifecycle
+        override func viewDidLoad() {
+            super.viewDidLoad()
+            NotificationCenter.default.addObserver(self, selector: #selector(updateChart(notification:)), name: .updateChart, object: nil)
+        }
+
+        // MARK: - Methods
+        @objc func updateChart(notification: Notification) {
+            let calorieChart = Chart(frame: chartView.frame)
+            var data: [Double] = []
+            for calorie in fetchedResultsController.fetchedObjects! {
+                data.append(Double(calorie.calorieAmount!) as! Double)
+            }
+            let series = ChartSeries(data)
+            series.area = true
+            calorieChart.add(series)
+            self.chartView.subviews.forEach({ $0.removeFromSuperview() })
+            self.chartView.addSubview(calorieChart)
```

Comment:
So here you are creating a new instance of your chart class each time you add a new calorie intake. You can just declare that calorieChart Globally and then clear the calorieCharts series property.

calorieChat.series = []

Reply:
You also have the exact same function as another student here...